src/main/java/edu/harvard/iq/dataverse/util/FileUtil.java

```diff
@@ -1091,8 +1091,13 @@ public static String generateOriginalExtension(String fileType) {
 
 			} 
 		} else {
-			//Remote file, trust supplier
-			finalType = suppliedContentType;
+			if(suppliedContentType==FileUtil.MIME_TYPE_UNDETERMINED_DEFAULT) {
+				finalType=determineFileTypeByExtension(fileName);
+				logger.fine("Determined type: " + finalType);
+			} else {
+			  //Remote file, trust supplier
+			  finalType = suppliedContentType;
```

Comment:
Do we care if the suppliedContentType is blank or null?

Reply:
That should already be being caught (e.g. in the EditDatafilesPage.handleExternalUpload method) but worth a safety check here probably. And we shouldn't replace the default with a null/blank response if determineFileTypeByExtension returns null/blank.

Reply:
If you make that change, I'll put this into QA. Thanks!